torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,513 @@
+        modality: list[str] = ["mask", "esa_wc", "viirs"],
+        split: str = "train",
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new MapInWild dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            modality: the modality to download. Choose from: "mask", "esa_wc",
+                "viirs", "s1", "s2_temporal_subset", "s2_[enter_season]".
+            split: one of "train", "val", or "test"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if ``split`` argument is invalid
+            ImportError: if pandas is not installed
+        """
+        assert split in ["train", "validation", "test"]
+
+        self.band_indices = torch.tensor(
+            [self.BAND_SETS["s2"].index(b) for b in self.BAND_SETS["s2"]]
+        ).long()
+
+        self.checksum = checksum
+        self.root = root
+        self.transforms = transforms
+        self.modality = modality
+        self.download = download
+        self._verify_split()
+
+        try:
+            import pandas as pd  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "pandas is not installed and is required to use this dataset"
+            )
+
+        split_dataframe = pd.read_csv(os.path.join(self.root, "split_IDs.csv"))
+        self.ids = split_dataframe[split].dropna().values.tolist()
+        self.ids = [int(i) for i in self.ids]
+
+        for modal in modality:
+            for modality_link in self.modality_urls[modal]:
+                self._verify(modality_link, self.md5s[os.path.split(modality_link)[1]])
+                if checksum:
+                    if not self._check_integrity(
+                        modality_link, self.md5s[os.path.split(modality_link)[1]]
+                    ):
+                        raise RuntimeError("Dataset not found or corrupted.")
+
+            if modal not in os.listdir(root) and len(self.modality_urls[modal]) == 2:
+                self.merge_parts(root, modal)  # same modality two times
+
+        if "mask" in self.modality:
+            self.modality.remove("mask")
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        self.list_modals = []
+        id = self.ids[index]
+
+        mask = self._load_raster(id, "mask")
+        mask[mask != 0] = 1
+
+        if "s2_temporal_subset" in self.modality:
```

Comment:
Couldn't all these if-statements be replaced with a for-loop? Something like:
```python
for mode in self.modality:
    data = self._load_raster(id, mode)
    if mode.startswith("s2_"):
        data = torch.index_select(data, dim=0, index=self.band_indices)
    self.list_modals.append(data)
```

Reply:
Indeed! Was planning to provide the user with a modality-specific band argument with these if-statements in the future but we can keep it like this for the first version, it is much more readable and clean.